pkg/sdkserver/sdkserver.go

```diff
@@ -547,21 +555,68 @@ func (s *SDKServer) stopReserveTimer() {
 // [Stage:Alpha]
 // [FeatureFlag:PlayerTesting]
 func (s *SDKServer) PlayerConnect(ctx context.Context, id *alpha.PlayerId) (*alpha.Empty, error) {
-	panic("implement me")
+	if !runtime.FeatureEnabled(runtime.FeaturePlayerTracking) {
+		return nil, errors.New(string(runtime.FeaturePlayerTracking) + " not enabled")
+	}
+	s.logger.WithField("PlayerId", id).Debug("Player Connected")
+
+	s.gsUpdateMutex.RLock()
```

Comment:
This looks like a potential race condition. You grab a read lock, drop it, and then conditionally grab a write lock depending on the result of the first read. I doubt there will be so much lock contention that just grabbing the write lock and doing the entire check and update inside of it will cause performance issues, but it will guarantee safety of mutations.

Reply:
I'm also wondering if using the standard lock and then defer unlock would be any different than explicitly unlocking. Does the enqueue call have the potential to block?

Reply:
> This looks like a potential race condition.

Well spotted. I agree, I will make this change here and below.

> I'm also wondering if using the standard lock and then defer unlock would be any different than explicitly unlocking. Does the enqueue call have the potential to block?

The enqueue could possibly result in a deadlock, since the sync function will try and lock the same mutex. So to be safe, we Unlock before we enqueue.